ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BlockHeaderValidator.java

```diff
@@ -99,6 +99,10 @@ private boolean applyRules(
     return parent;
   }
 
+  public void addRule(final AttachedBlockHeaderValidationRule<C> rule) {
```

Comment:
This makes the BlockHeaderValidator mutable, and IMHO should not be done.  Note we have a builder that has the exact same method, so instead of mutating a produced BlockHeaderValidator we should hook into wherever the builder is being created.

Reply:
Done

Reply:
We still need to delete this method.